File: src/main/java/dev/zprestige/fire/module/client/clickgui/TickListener.java
package dev.zprestige.fire.module.client.clickgui;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.TickEvent;
import dev.zprestige.fire.ui.panel.PanelScreen;

public class TickListener extends EventListener<TickEvent, ClickGui> {

    public TickListener(final ClickGui panel) {
        super(TickEvent.class, panel);
    }

    @Override
    public void invoke(final Object object) {
        if (!(mc.currentScreen instanceof PanelScreen)) {
            module.disableModule();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/misc/autorespawn/TickListener.kt
package dev.zprestige.fire.module.misc.autorespawn

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.TickEvent
import net.minecraft.client.gui.GuiGameOver

class TickListener(autoRespawn: AutoRespawn) : EventListener<TickEvent, AutoRespawn>(TickEvent::class.java, autoRespawn) {

    override fun invoke(e: Any) {
        if (mc.currentScreen is GuiGameOver) {
            mc.player.respawnPlayer()
            mc.displayGuiScreen(null)
        }
    }
}
